cmd/podman/create_cli.go

```diff
@@ -74,6 +75,77 @@ func addWarning(warnings []string, msg string) []string {
 	return append(warnings, msg)
 }
 
+// Format supported.
+// podman run --mount type=bind,src=/etc/resolv.conf,target=/etc/resolv.conf ...
+// podman run --mount type=tmpfs,target=/dev/shm ..
+func parseMounts(mounts []string) ([]spec.Mount, error) {
+	var mountList []spec.Mount
+	errInvalidSyntax := errors.Errorf("incorrect mount format : should be --mount type=<bind|tmpfs>,src=<host-dir>,target=<ctr-dir>,[options]")
+	for _, mount := range mounts {
+		var tokenCount int
+		var mountInfo spec.Mount
+
+		arr := strings.SplitN(mount, ",", 2)
+		if len(arr) < 2 {
+			return nil, errInvalidSyntax
+		}
+		kv := strings.Split(arr[0], "=")
```

Comment:
probably should add a check for 'type' in kv[0] here.  I know we validate that we get bind/tmpfs, but we don't see if they gave us the right first word that I see and we probably should.

Reply:
Fixed